userspace/otpilot/src/main.rs

```diff
@@ -183,9 +199,24 @@ fn run() -> TockResult<()> {
             console_reader::get().allow_read(1)?;
         }
 
+        if gpio::get().have_events() {
+            match gpio_processor.process_gpio_events() {
+                Ok(()) => {}
+                Err(_) => {
+                    // Ignore error from writeln. There's nothing we can do here anyway.
+                    let _ = writeln!(console, "Device: Error processing GPIO event.");
+                }
+            }
+        }
+
         if alarm::get().is_expired() {
-            writeln!(console, "Alarm expired.")?;
-            alarm::get().clear()?;
+            match gpio_processor.alarm_expired() {
+                Ok(()) => {}
+                Err(_) => {
+                    // Ignore error from writeln. There's nothing we can do here anyway.
+                    let _ = writeln!(console, "Alarm: Error processing event.");
```

Comment:
"Error processing event."?  Is that right?

Reply:
Yeah - it's an alarm event. I think that's ok.